cpp/src/parquet/page_index.cc

```diff
@@ -636,6 +691,16 @@ class OffsetIndexBuilderImpl final : public OffsetIndexBuilder {
             page_location.__set_offset(page_location.offset + final_position);
           }
         }
+
+        /// Finalize unencoded_byte_array_data_bytes and make sure page sizes match.
+        if (offset_index_.page_locations.size() ==
+            offset_index_.unencoded_byte_array_data_bytes.size()) {
+          offset_index_.__isset.unencoded_byte_array_data_bytes = true;
+        } else {
+          /// Discard unencoded_byte_array_data_bytes if its size is abnormal.
+          offset_index_.unencoded_byte_array_data_bytes.clear();
```

Comment:
same question as above  about exception/log vs silent discard here

Reply:
Good suggestion. I have switched to exception.